### PR TITLE
#84 / SDCISA-3814 Test stability improvements

### PR DESCRIPTION
**Pull request by Kusig**

### CODE REVIEW

File: src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java [hunk 1]
```diff
@@ -2157,37 +2092,140 @@ public void getQueueSpeedMatchingPartlyFilter(TestContext context) throws Except
...
+     * @param queueSize             The size of the particular expected queue at position=element
+     * @param queueFailures         The failures of the particular expected queue at
+     *                              position=element
+     * @param queueSlowdownTime     The slowDownTime of the particular expected queue at
+     *                              position=element
+     * @param queueBackpressureTime The backpressureTime of the particular expected queue at
+     *                              position=element
+     * @oaram filter                The filter to be applied to the statistic queues request
+     */
+    private void assertQueueState(TestContext context, String filter, Integer queuesSize,
```
Comment (dominik-cnx) [inline src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:2117]: Minor, maybe annotate the params that can be null with @Nullable

File: src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java [hunk 2]
```diff
@@ -1979,87 +1964,46 @@ public void getStatisticsSlowDown(TestContext context) throws Exception {
...
-                .then()
-                .assertThat().statusCode(200)
-                .extract().path("queues");
-            context.assertTrue(response.size() == 1);
-            context.assertTrue(response.get(0).get("name").equals("stat_b"));
-            context.assertTrue(response.get(0).get("size").equals(1));
-            context.assertTrue(((int) response.get(0).get("failures")) >= 1);
-            context.assertTrue(((int) response.get(0).get("slowdownTime")) >= 0);
-
-            // now we wait just some time up until the failures and slowdowns are applied
```
Comment (dominik-cnx) [inline src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:1992]: It looks like this was accidentally removed?

Comment (Kusig) [reply src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:1992]: The checks are now part of the assertQueueState method itself and therefore not needed anymore.  As we now do wait up until the requested state is reached, we could check for particular specific foreseen situation.

File: src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java (hunk 1 above)
Comment (Kusig) [reply src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:2117]: Will do.

File: src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java (hunk 2 above)
Comment (dominik-cnx) [reply src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:1992]: Hmm yes, but the 5 sec sleep on line #1993 is gone now but still mentioned on line #1980 (left)/ #1965 (right) ?

Comment (Kusig) [reply src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:1992]: Aha, that's what you mean.  The strategy changed but the 5 seconds are still valid as this is now the maximum time we wait up until the expected situation must be fullfilled.  There is no explicit wait anymore but a constantly check during this time.

Comment (dominik-cnx) [reply src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java:1992]: OK, thx for the clarification

**Review by dominik-cnx — APPROVED** [review]

**Pull request merged**